langchain/vectorstores/base.py

```diff
@@ -324,13 +347,29 @@ def validate_search_type(cls, values: Dict) -> Dict:
         """Validate search type."""
         if "search_type" in values:
             search_type = values["search_type"]
-            if search_type not in ("similarity", "mmr"):
+            if search_type not in ("similarity", "similarity_score_threshold", "mmr"):
                 raise ValueError(f"search_type of {search_type} not allowed.")
+            if search_type == "similarity_score_threshold":
+                score_threshold = values["search_kwargs"].get("score_threshold")
+                if (score_threshold is None) or (
+                    not isinstance(score_threshold, float)
+                ):
```

Comment:
Why separate condition for `None`, if isinstance for float is used then we should be able to validate it's None or not

```
>>> score_threshold  = None
>>> isinstance(score_threshold, float)
False
>>> score_threshold  = 0.15
>>> isinstance(score_threshold, float)
True
```

Reply:
makes sense! will make the change